src/Memoize.jl

```diff
@@ -78,15 +78,10 @@ macro memoize(args...)
 end
 
 macro clear_cache(f)
-    fcachename = Symbol("##", f, "_memoized_cache")
-    mod = __module__
-    fcache = getfield(mod, fcachename)
-
-    function clear_cache!(x::IdDict)
-        foreach(k -> delete!(fcache, k), keys(fcache))
-    end
-
-    Core.eval(mod, :($(clear_cache!(fcache))))
+    esc(quote
+        fcachename = Symbol("##", $f, "_memoized_cache")
+        Base.empty!(eval(fcachename))
+    end)
```

Comment:
```suggestion
    fcachename = Symbol("##", $f, "_memoized_cache")
    esc(quote
        Base.empty!($fcachename)
    end)
```
The less `eval` the better...